src/geojson-layer.ts

```diff
@@ -169,11 +214,31 @@ export default class GeoJSONLayer extends React.Component<Props, {}> {
           const layoutDiff = diff(this.props[prop], props[prop]);
 
           Object.keys(layoutDiff).forEach(key => {
-            map.setLayoutProperty(`${this.id}-${type}`, key, layoutDiff[key]);
+            map.setLayoutProperty(this.buildLayerId(type), key, layoutDiff[key]);
           });
         }
       }
     }
+
+    for (const type in typeToLayerLUT) {
```

Comment:
Line 209 we already iterate over the `typeToLayerLUT`, if you could insert your block in there that would be better. Also it is a technical debt I would rather use `Object.keys().forEach` too